pkg/fuzzer/queue/queue.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package queue
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+
+	"github.com/google/syzkaller/pkg/ipc"
+	"github.com/google/syzkaller/pkg/signal"
+	"github.com/google/syzkaller/pkg/stats"
+	"github.com/google/syzkaller/prog"
+)
+
+type Request struct {
+	Prog       *prog.Prog
+	NeedSignal SignalType
+	NeedCover  bool
```

Comment:
VM checking will need to control env/exec flags precisely, few bools won't be enough for full flexibility:
https://github.com/dvyukov/syzkaller/blob/f4b95d5d44a9157fa9baf1e7017a52bdfd02e12b/pkg/vminfo/features.go#L109-L150

syz-ci reproducer testing and runtest will also need to submit binary C programs.